# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.cpp
# Commit: 25281e1775c9
# Full Hash: 25281e1775c9bd2c600fc897ce877d7da55ce3d8
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-30 03:48:30
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 4: Untie frame iteration from Frame::tls. r=lth
#   
#   Here we replace usage of Frame::tls in frame iteration with GetNearestEffectiveTls.
#   We also maintain current tls for frame iteration object to not to call GetNearestEffectiveTls everytime.
#   
# ==============================================================================

diff -r 82fbedcdde75 -r 25281e1775c9 js/src/jit/MacroAssembler.cpp
--- a/js/src/jit/MacroAssembler.cpp	Fri Oct 23 18:17:37 2020 +0000
+++ b/js/src/jit/MacroAssembler.cpp	Fri Oct 23 18:17:37 2020 +0000
@@ -4657,19 +4657,22 @@
 template void AutoGenericRegisterScope<Register>::reacquire();
 template void AutoGenericRegisterScope<FloatRegister>::reacquire();
 
-wasm::TlsData* ExtractCallerTlsFromFrameWithTls(wasm::Frame* fp) {
-  return *reinterpret_cast<wasm::TlsData**>(
-      reinterpret_cast<uint8_t*>(fp) + sizeof(wasm::Frame) + ShadowStackSpace +
-      wasm::FrameWithTls::callerTLSOffset());
-}
-
-wasm::TlsData* ExtractCalleeTlsFromFrameWithTls(wasm::Frame* fp) {
-  return *reinterpret_cast<wasm::TlsData**>(
-      reinterpret_cast<uint8_t*>(fp) + sizeof(wasm::Frame) + ShadowStackSpace +
-      wasm::FrameWithTls::calleeTLSOffset());
-}
-
 #endif  // DEBUG
 
 }  // namespace jit
+
+namespace wasm {
+TlsData* ExtractCallerTlsFromFrameWithTls(Frame* fp) {
+  return *reinterpret_cast<TlsData**>(reinterpret_cast<uint8_t*>(fp) +
+                                      sizeof(Frame) + ShadowStackSpace +
+                                      FrameWithTls::callerTLSOffset());
+}
+
+TlsData* ExtractCalleeTlsFromFrameWithTls(Frame* fp) {
+  return *reinterpret_cast<TlsData**>(reinterpret_cast<uint8_t*>(fp) +
+                                      sizeof(Frame) + ShadowStackSpace +
+                                      FrameWithTls::calleeTLSOffset());
+}
+}  // namespace wasm
+
 }  // namespace js